server/src/main/java/org/opensearch/cluster/routing/WeightedRouting.java

```diff
@@ -40,6 +46,10 @@ public WeightedRouting(StreamInput in) throws IOException {
         weights = (Map<String, Double>) in.readGenericValue();
     }
 
+    public boolean isSet() {
+        return (!this.attributeName.isEmpty() && !this.weights.isEmpty());
+    }
```

Comment:
When is this needed?

Reply:
During `delete`  api call, weighted routing is set empty since we are retaining version in metadata. This is to check for empty weights